jumanji/environments/packing/bin_pack/generator.py

```diff
@@ -875,3 +880,74 @@ def body_fn(
         )
 
         return items_spaces, items_mask
+
+
+class ConstrainedToyGenerator(ToyGenerator):
+    def __init__(self) -> None:
+        super().__init__()
+
+    def _generate_solved_instance(self, key: chex.PRNGKey) -> State:
+        solution = super()._generate_solved_instance(key)
+        x_len, y_len, z_len = (
+            solution.items.x_len,
+            solution.items.y_len,
+            solution.items.z_len,
+        )
+        solution.items = Item(
+            x_len=jnp.array([x_len, x_len, y_len, y_len, z_len, z_len]),
+            y_len=jnp.array([y_len, z_len, x_len, z_len, y_len, x_len]),
+            z_len=jnp.array([z_len, y_len, z_len, x_len, x_len, y_len]),
+        )
+        solution.items_mask = jnp.array(
+            [
+                [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+                [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+                [0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 1, 0],
```

Comment:
Is this because eg. the first item can only be placed in the first 2 orientations?

Reply:
Yes ,  I will add a comment explaining it.

Reply:
Yes, I added a comment explaining this.